File: event-service/src/main/java/com/greenlight/event/domain/Event.java
package com.greenlight.event.domain;

import lombok.Data;
import lombok.RequiredArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.geo.Point;

import java.time.LocalDateTime;

@Data
@RequiredArgsConstructor
public class Event {

  @Id
  private final Long id;

  private String title;
  private String description;

  private EventStatus status;
  private Point geoLocation;

  private final LocalDateTime createdAt;
  private LocalDateTime updatedAt;
  private LocalDateTime finishedAt;
  private Long authorId;
}

File: event-service/src/main/kotlin/com/greenlight/event/domain/Event.kt
package com.greenlight.event.domain

import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.DBRef
import org.springframework.data.mongodb.core.mapping.Document
import java.time.LocalDateTime

@Document
data class Event(
    @Id
    val id: Long? = null,

    var title: String,

    var description: String,

    @DBRef
    var status: EventStatus? = null,

    var geoLocation: Point,

    val createdAt: LocalDateTime = LocalDateTime.now(),

    val startsAt: LocalDateTime,

    var updatedAt: LocalDateTime? = null,

    val finishedAt: LocalDateTime? = null,

    var authorId: Long? = null,

    var peopleRequired: Int
)